components/http-accept.js

```diff
@@ -61,30 +65,28 @@ function handle(pair){
                 outPorts.accepted.send(pair);
                 outPorts.accepted.disconnect();
             }
-            outPorts.out.send(pair.req.body);
-            outPorts.out.disconnect();
+            return pair.req.body;
         } else {
-            body(pair.req, {
-                limit: this.limit,
-                encoding: this.encoding
-            }, function(err, body){
-                if (err) {
-                    if (outPorts.rejected.isAttached()) {
-                        pair.res.writeHead(413);
-                        pair.res.write(err.message);
-                        outPorts.rejected.send(pair);
-                        outPorts.rejected.disconnect();
-                    }
-                } else {
-                    if (outPorts.accepted.isAttached()) {
-                        pair.res.writeHead(202);
-                        pair.res.write("Accepted\n");
-                        outPorts.accepted.send(pair);
-                        outPorts.accepted.disconnect();
-                    }
-                    outPorts.out.send(body);
-                    outPorts.out.disconnect();
+            var charset = typer.parse(pair.req.headers['content-type']).parameters.charset;
+            return getRawBody(pair.req, {
+                limit: this.limit || '1mb',
+                encoding: charset || this.encoding || 'utf8'
+            }).then(function(body){
+                if (outPorts.accepted.isAttached()) {
+                    pair.res.writeHead(202);
+                    pair.res.write("Accepted\n");
+                    outPorts.accepted.send(pair);
+                    outPorts.accepted.disconnect();
+                }
+                return body;
+            }, function(err){
+                if (outPorts.rejected.isAttached()) {
+                    pair.res.writeHead(413);
```

Comment:
You are using hard coded values like 202 or 413 for your header status.  Would rather see something like HttpStatus.ACCEPTED or HttpStatus.REQUEST_TOO_LONG since it's more clear without having to look them up.